Sources/Controllers/MessagesViewController+Keyboard.swift

```diff
@@ -122,8 +122,8 @@ internal extension MessagesViewController {
     }
 
     internal func requiredInitialScrollViewBottomInset() -> CGFloat {
-        guard let inputAccessoryView = inputAccessoryView else { return 0 }
-        return max(0, inputAccessoryView.frame.height + additionalBottomInset - automaticallyAddedBottomInset)
+        let inputAccessoryViewHeight = inputAccessoryView?.frame.height ?? 0
```

Comment:
Personally I'm not a fan of optional chaining anymore, it can and has caused soft-failures to slip past unnoticed. Usually in a guard let there is a log statement before returning so that we can be notified that the execution went down an 'unhappy path'. However, I'm not aware of any logging standards in the MessageKit framework thus far. What if we did it like this:

```swift
    internal func requiredInitialScrollViewBottomInset() -> CGFloat {
        var adjustedBottomInset = additionalBottomInset - automaticallyAddedBottomInset
        if let inputAccessoryView = inputAccessoryView, inputAccessoryView.frame.height > 0 {
            adjustedBottomInset += inputAccessoryView.frame.height
        }
        return max(0, adjustedBottomInset)
    }
```

This leaves room to log something within an 'else' statement in the future. Execution can continue if the conditional unwrapping fails, so we go with `if let` instead of `guard`. This also prevents us from having near-duplicate code in what would have been the guard's early return statement.